scripts/fetch-icon.js

```diff
@@ -0,0 +1,107 @@
+#!/usr/bin/env node
+
+const inquirer = require('inquirer');
+const xRay = require('x-ray')();
+const fetch = require('node-fetch');
+const { mkdirSync, writeFileSync, existsSync } = require('fs');
+const { join, extname } = require('path');
+const url = require('url');
+const makeDriver = require('request-x-ray');
+
+// mobile user agent because some sites only serve manifest/pwa for mobile
+const headers = {
+  'User-Agent':
+    'Mozilla/5.0 (Linux; Android 8.0; Pixel 2 Build/OPD3.170816.012) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/67.0.3396.99 Mobile Safari/537.36'
+};
+
+xRay.driver(makeDriver({ headers }));
+
+/**
+ *
+ * @param {string} size the size of the image, like "144x144"
+ * @returns the first dimension, like 144
+ */
+const parseSize = size => parseInt(size.split('x')[0]);
+
+/**
+ *
+ * @param {{sizes: string, src: string}[]} icons
+ */
+const findLargestIcon = icons => {
+  const icon = icons.reduce((largestIcon, icon) => {
+    if (
+      parseSize(icon.sizes) >
+      (largestIcon.sizes ? parseSize(largestIcon.sizes) : 0)
+    ) {
+      return icon;
+    }
+    return largestIcon;
+  });
+  return { icon: icon.src, size: icon.sizes };
+};
+
+/**
+ * @param {string} u the url to retrieve the name from
+ * @returns {string|undefined}
+ */
+const simplifyUrl = u => {
+  const { hostname } = url.parse(u);
+  if (hostname === null) {
+    return undefined;
+  }
+  return hostname
+    .replace(/\.[^.]*$/, '') // remove tld (last dot and everything after it)
+    .replace(/^.*\./, '') // remove subdomains (everything up to the last dot)
+    .toLowerCase();
+};
+
+module.exports = simplifyUrl;
+
+const main = async () => {
+  const { appUrl } = await inquirer.prompt([
+    {
+      name: 'appUrl',
+      type: 'input',
+      message: 'Link to a page in the PWA'
+    }
+  ]);
+  const manifestLink = await xRay(appUrl, 'link[rel="manifest"]@href')
+    .then(link => url.resolve(appUrl, link))
+    .catch(err => {
+      throw new Error(`Unable to find manifest for ${appUrl}`);
```

Comment:
Is there any way to make the output a bit nicer when it fails to find the manifest?

This might seem scary to some. 🙈 

```
? Link to a page in the PWA https://google.com
Error finding icon for PWA
Error: Unable to find manifest for https://google.com
    at xRay.then.catch.err (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/scripts/fetch-icon.js:71:13)
    at tryCatcher (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/util.js:16:23)
    at Promise._settlePromiseFromHandler (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/promise.js:512:31)
    at Promise._settlePromise (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/promise.js:569:18)
    at Promise._settlePromise0 (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/promise.js:614:10)
    at Promise._settlePromises (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/promise.js:689:18)
    at Async._drainQueue (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/async.js:133:16)
    at Async._drainQueues (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/async.js:143:10)
    at Immediate.Async.drainQueues (/Users/Rodriguez/Development/projects/cloudfour/repos/pwastats/node_modules/bluebird/js/release/async.js:17:14)
    at runCallback (timers.js:810:20)
    at tryOnImmediate (timers.js:768:5)
    at processImmediate [as _immediateCallback] (timers.js:745:5)
```
If you disagree or it's not possible, then this is fine. :)

Reply:
I could exclude the stack trace, but I think the output:

- Is understandable by someone less technical:
  ```
  Error finding icon for PWA
  Error: Unable to find manifest for https://google.com
  ```

- Would help if someone was trying to debug a problem

What if I make the error in red, and the stack trace in a faded grey, so it stands out less and is less scary for less technical people?

Reply:
> What if I make the error in red, and the stack trace in a faded grey, so it stands out less and is less scary for less technical people?

Yep! This works! :)

Reply:
Fixed ✔️

<img width="960" alt="screen shot 2018-07-17 at 11 11 10 am" src="https://user-images.githubusercontent.com/13206945/42837216-4907427e-89b2-11e8-99b8-54b2725efacd.png">